### PR TITLE
Feature/fks 101

### PR DESCRIPTION
**Pull request by Evil00787**
Added child plan widget and sample screen

### CODE REVIEW

File: lib/pages/child/panel_page.dart [hunk 1]
```diff
@@ -1,27 +1,113 @@
...
+            physics: BouncingScrollPhysics(),
+            children: <Widget>[
+              Padding(
+                  padding: EdgeInsets.only(
+                      bottom: AppBoxProperties.sectionPadding,
+                      right: AppBoxProperties.screenEdgePadding,
+                      left: AppBoxProperties.screenEdgePadding),
+                  child: Text(
+                      '${AppLocales.of(context).translate("page.childSection.panel.content.inProgress")} ',
+                      style: Theme.of(context).textTheme.headline1)),
```
Comment (stasgora) [inline lib/pages/child/panel_page.dart:36]: 1. This inconsistent with what Mikołaj has (font size)
2. I already asked him to make a widget out of that so you might want to wait for his merge

File: lib/widgets/plan_widget.dart [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                                      size: tasksIconSize,
+                                    )),
+                                Text(
+                                    ' ${AppLocales.of(context).translate("page.childSection.panel.content.tasks", {
+                                      'NUM_TASKS': this.tasksCount
+                                    })}',
+                                    style: blueTextStyle)
+                              ],
+                            )
+                          ])),
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:100]: This is another potential candidate for a widget on Mikołaj side

File: lib/pages/child/panel_page.dart [hunk 3]
```diff
@@ -1,27 +1,113 @@
...
+                                  Padding(padding: EdgeInsets.only(right: AppBoxProperties.buttonIconPadding), child:Icon(
+                                    Icons.calendar_today,
+                                    color: Colors.white,
+                                  )),
+                                  Text(
+                                    '${AppLocales.of(context).translate("page.childSection.panel.content.futurePlans")} ',
+                                    style: Theme.of(context).textTheme.button,
+                                  )
+                                ],
+                              ))))
```
Comment (stasgora) [inline lib/pages/child/panel_page.dart:105]: This type of button with icon is used at least twice - make a widget out of it

File: lib/pages/child/panel_page.dart [hunk 4]
```diff
@@ -1,27 +1,113 @@
...
-			),
-			bottomNavigationBar: AppNavigationBar.childPage(currentIndex: 0)
-		);
-	}
+  // styling vars
+  final Color futurePlansColor = Colors.amber;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Comment (stasgora) [inline lib/pages/child/panel_page.dart:20]: Try to break functions this long into logical smaller parts. Here the current, todays, and button for example

File: lib/widgets/plan_widget.dart [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:fokus/utils/app_locales.dart';
+import 'package:fokus/utils/theme_config.dart';
+
+class PlanWidget extends StatelessWidget {
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:7]: Trying to standardize naming conventions a little, maybe a PlanCard?

File: lib/widgets/plan_widget.dart [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  final TextStyle blueTextStyle =
+      new TextStyle(fontSize: 16, color: Colors.blue);
+
+  PlanWidget(
+      {@required this.planName,
+      @required this.isActive,
+      @required this.isInProgress,
+      @required this.tasksCount,
+      @required this.details,
+      this.buttonAction = null,
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:36]: https://dart-lang.github.io/linter/lints/avoid_init_to_null.html

File: lib/widgets/plan_widget.dart [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  final Color inProgressButtonColor = Colors.amber;
+  final Icon notInProgressIcon = new Icon(Icons.play_arrow);
+  final Icon inProgressIcon = new Icon(Icons.launch);
+  final TextStyle blueTextStyle =
+      new TextStyle(fontSize: 16, color: Colors.blue);
+
+  PlanWidget(
+      {@required this.planName,
+      @required this.isActive,
+      @required this.isInProgress,
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:33]: This seems kind of excessive together with progressPercentage param. As plan in progress can still have 0% completion we could take a nullable int as progress. That way:
- the default (not passed) would mean it is not in progress
- any passed number means it is

File: lib/widgets/plan_widget.dart [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  final Color inactiveProgressBar = Color.fromARGB(255, 240, 240, 240);
+  final Color inProgressButtonColor = Colors.amber;
+  final Icon notInProgressIcon = new Icon(Icons.play_arrow);
+  final Icon inProgressIcon = new Icon(Icons.launch);
+  final TextStyle blueTextStyle =
+      new TextStyle(fontSize: 16, color: Colors.blue);
+
+  PlanWidget(
+      {@required this.planName,
+      @required this.isActive,
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:32]: Maybe default to false?

File: lib/widgets/plan_widget.dart [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+                          ])),
+                  Row(
+                    children: <Widget>[
+                      Expanded(child: addProgressIndicatorIfAvailable())
+                    ],
+                  )
+                ],
+              ),
+            ),
+            ClipRRect(
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:109]: This ClipRect is not needed. FlatButton has its own shape param that can take RoundedRectangleBorder, same with the other button

File: lib/widgets/plan_widget.dart [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        : Icon(notInProgressIcon.icon,
+            color: buttonIconColor, size: playButtonSize);
+  }
+
+  Widget addProgressIndicatorIfAvailable() {
+    if (!isInProgress)
+      return Container();
+    else
+      return ClipRRect(
+          borderRadius: BorderRadius.only(bottomLeft: Radius.circular(5)),
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:149]: Not using the roundedCornersRadius here?

File: lib/widgets/plan_widget.dart [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+  final double tasksIconSize = 26.0;
+  final double progressIndicatorHeight = 15.0;
+  final double futurePlansButtonWidth = 80.0;
+  final int titleMaxLines = 3;
+  final Color activeButtonColor = Colors.teal;
+  final Color buttonIconColor = Colors.white;
+  final Color disabledButtonColor = Color.fromARGB(255, 200, 200, 200);
+  final Color inactiveProgressBar = Color.fromARGB(255, 240, 240, 240);
+  final Color inProgressButtonColor = Colors.amber;
+  final Icon notInProgressIcon = new Icon(Icons.play_arrow);
```
Comment (stasgora) [inline lib/widgets/plan_widget.dart:25]: You know that the _new_ keyword is optional in Dart?

File: lib/widgets/rounded_button.dart [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                      child: Icon(
+                        this.iconData,
+                        color: Colors.white,
+                      )),
+                Text(
+                  this.text,
+                  style: Theme.of(context).textTheme.button,
+                )
+              ],
+            )));
```
Comment (stasgora) [inline lib/widgets/rounded_button.dart:43]: The formatting is still not great :( It's a shame it doesn't handle it automatically. I mean it's Google, it shouldn't be to hard for them.

File: lib/pages/child/panel_page.dart [hunk 13]
```diff
@@ -1,36 +1,107 @@
...
+              return Expanded(
+                  child: Center(child: CircularProgressIndicator()));
+            },
+          ),
+          Row(
+						mainAxisAlignment: MainAxisAlignment.end,
+            children: <Widget>[RoundedButton(
+              iconData: Icons.calendar_today,
+              text:
+                  '${AppLocales.of(context).translate("page.childSection.panel.content.futurePlans")} ',
```
Comment (stasgora) [inline lib/pages/child/panel_page.dart:49]: Be consistent with the _pageKey use

File: lib/widgets/app_header.dart [hunk 14]
```diff
@@ -15,39 +15,49 @@ import 'package:fokus/widgets/attribute_chip.dart';
 enum AppHeaderType { greetings, normal }
 
 class HeaderActionButton {
-	final IconData icon;
```
[the remark this answers is not in the document]
Comment (Evil00787) [reply lib/widgets/app_header.dart:18]: It wasn't 🤔

File: lib/widgets/rounded_button.dart (hunk 12 above)
Comment (Evil00787) [reply lib/widgets/rounded_button.dart:43]: I will try to adjust my formatting to yours 😓

**Pull request merged**